sdks/python/apache_beam/io/hadoopfilesystem_test.py

```diff
@@ -153,7 +153,7 @@ def delete(self, path, recursive=True):
 
     _ = self.status(path)
 
-    for filepath in self.files.keys():  # pylint: disable=consider-iterating-dictionary
+    for filepath in list(self.files):
```

Comment:
Any specific reason for not using `six` library for porting? I see our dependencies are using six. 
In case we are using six this could be something like this:
```
 import six
..
..
for filepath in six.iterkeys(self.files):
```

Reply:
This could be also efficient in terms of memory.

Reply:
We prefer to use idiomatic Python 3, instead of a back-port code, see the conversion proposal in: https://s.apache.org/beam-python-3. When there are concerns with Python 3 code being memory-inefficient we use future.utils.iterkeys() of branch implementation based on version.